evmrpc/tx.go

```diff
@@ -85,6 +87,39 @@ func (t *TransactionAPI) GetTransactionByBlockHashAndIndex(ctx context.Context,
 	return t.getTransactionWithBlock(block, index)
 }
 
+func (t *TransactionAPI) GetPendingNonces(ctx context.Context, addr common.Address) (string, error) {
+	sdkCtx := t.ctxProvider(LatestCtxHeight)
+	nonces := []int{}
+	for page := 1; page <= UnconfirmedTxQueryMaxPage; page++ {
+		res, err := t.tmClient.UnconfirmedTxs(ctx, &page, nil)
+		if err != nil {
+			return "", err
+		}
+		if len(res.Txs) == 0 {
+			break
+		}
+		for _, tx := range res.Txs {
+			etx := getEthTxForTxBz(tx, t.txConfig.TxDecoder())
+			if etx != nil {
+				signer := ethtypes.MakeSigner(
+					t.keeper.GetChainConfig(sdkCtx).EthereumConfig(t.keeper.ChainID(sdkCtx)),
+					big.NewInt(sdkCtx.BlockHeight()),
+					uint64(sdkCtx.BlockTime().Second()),
+				)
+				from, _ := ethtypes.Sender(signer, etx)
+				if from == addr {
+					nonces = append(nonces, int(etx.Nonce()))
+				}
+			}
+		}
+		if page*30 >= res.Total {
```

Comment:
out of curiosity - what is the reasoning for this limit?

Reply:
this is to check if this is the last page (30 is the default page size; I've changed it to reference the actual constant)